labman/db/support_files/db_patch_manual.sql

```diff
@@ -3525,3 +3525,81 @@ INSERT INTO qiita.shotgun_combo_primer_set (shotgun_primer_set_id, i5_primer_set
 -- Trigger: When inserting in sample composition, check that if sample_composition_type_id
 -- points to experimental sample then sample_id should be provided, otherwise it should be
 -- always null
+
+-- Creating a function to fast retrieve the study titles from a plate
+CREATE OR REPLACE FUNCTION qiita.get_plate_studies(in_plate_id BIGINT) RETURNS varchar[] AS $$
+DECLARE
+    plate_type  VARCHAR;
+BEGIN
+    SELECT DISTINCT description INTO plate_type
+        FROM qiita.plate
+            JOIN qiita.well USING (plate_id)
+            JOIN qiita.composition USING (container_id)
+            JOIN qiita.composition_type USING (composition_type_id)
+        WHERE plate_id = in_plate_id;
+
+    IF plate_type = 'sample' THEN
+        RETURN (SELECT array_agg(DISTINCT study_title)
+                FROM qiita.study
+                    JOIN qiita.study_sample USING (study_id)
+                    JOIN qiita.sample_composition USING (sample_id)
+                    JOIN qiita.composition USING (composition_id)
+                    JOIN qiita.well USING (container_id)
+                WHERE plate_id = in_plate_id);
+    ELSIF plate_type = 'gDNA' THEN
+        RETURN (SELECT array_agg(DISTINCT study_title)
+                FROM qiita.study
+                    JOIN qiita.study_sample USING (study_id)
+                    JOIN qiita.sample_composition USING (sample_id)
+                    JOIN qiita.gdna_composition gdna USING (sample_composition_id)
```

Comment:
Seems unfortunate that you need to have this big series of `ELSIF` statements. I assume it's because these trace back to unique tables per `plate_type`?

Reply:
Yes, It could've been done by creating functions for each plate type, but that would result on multiple queries that, although they're all happening at once in the DB server, this way is still faster cause it is specialized for each plate type, performing a single query per plate rather than multiple queries per plate to perform the traceback.